README.md

```diff
@@ -69,6 +69,63 @@ FullName.propTypes = {
 register(FullName, 'full-name');
 ```
 
+### API
+
+### register(Component, tagName, props?, options?)
+
+Returns a Custom Element.
+
+#### Component
+
+Type: `object`
+
+#### tagName
+
+Type: `string`
+
+#### props?
+
+Type: `Array<string>`
+
+#### options
+
+Type: `boolean | object`
+Default: `false`
+
+#### options.shadow
+
+Type: `boolean`
+Default: `false`
+
+Attaches a Shadow DOM instance to your Custom Element. Remember that global styles won't affect the Custom Element unless you inject (see `options.injectGlobalStyles`) or somehow import them from the component itself.
+
+#### options.injectGlobalStyles
```

Comment:
I wonder if we can slim down the user-facing API surface area. My gut tells me that we can combine the solutions:

- `options.injectGlobalStyles = true` is the same as `injectGlobalStyles.selector = '*'`
- We could eliminate `injectGlobalStyles.target` by looping over `document.styleSheets` directly. It includes both inline style tags and sheets added via a `link` element. Each sheet has a pointer to the node it was created by and we can match against that.

I'd love to start simple and remove `filter` too. Whilst there are cases in theory where a query selector may not suffice, I can't come up with realistic real world examples where the selector approach falls short. Checking popular CSS-in-JS libs, they all mark their own stylesheets with a special attribute. For CSS-modules the user usually has tight control over naming the assets and we can match on that.

Since there all sheets are inherently global, unless bound to a shadow root, we can drop the `Global` part of the variable name, imo.